subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/buildcache/TaskOutputCachingJavaPerformanceTest.groovy

```diff
@@ -55,6 +58,31 @@ class TaskOutputCachingJavaPerformanceTest extends AbstractTaskOutputCachingPerf
         result.assertCurrentVersionHasNotRegressed()
     }
 
+    @RunFor(
+        @Scenario(type = PER_DAY, operatingSystems = [LINUX], testProjects = ["smallJavaMultiProjectManyExternalDependencies"])
+    )
+    def "clean check on ephemeral ci"() {
+        runner.cleanTasks = ["clean"]
+        runner.tasksToRun = ["check"]
+        protocol = "http"
```

Comment:
🤔 Is there a reason why you can't use the local cache here? I'd expect that using the local build cache would make the test more stable.

Reply:
Mostly because it's closer to the real scenario - You don't have a local cache on ephemeral CI agents.

Reply:
It would be good to document this in a comment.